WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift

```diff
@@ -159,12 +161,23 @@ private extension ApplicationPasswordAuthorizationWebViewController {
             return
         }
         let request = URLRequest(url: urlWithQueries)
+        self.authorizationURL = request.url?.absoluteString
         webView.load(request)
-        analytics.track(.applicationPasswordAuthorizationWebViewShown)
     }
 
-    func handleAuthorizationResponse(with url: URL?) {
-        guard let url, url.absoluteString.hasPrefix(Constants.successURL) else {
+    func handleWebViewURL(_ url: URL?) {
+        guard let url else {
+            return
+        }
+        if url.absoluteString == authorizationURL {
+            analytics.track(event: .ApplicationPasswordAuthorization.webViewShown(step: webViewInitialLoad ? .initial : .authorization))
+            webViewInitialLoad = false
+        } else if url.absoluteString.contains(Constants.Query.redirect) {
+            analytics.track(event: .ApplicationPasswordAuthorization.webViewShown(step: .login))
+        }
```

Comment:
nit: this logic seems to be based on a few assumptions briefly mentioned in the PR description, which I don't have the most thorough understanding (like why having a `redirect_to` in the URL means it's the login step, what if the store URL also contains `redirect_to` by any rare chance) 😅 maybe a comment on these assumptions can be helpful for anyone new to this code.

there's another assumption that `url.absoluteString == authorizationURL` always happens before `url.absoluteString.contains(Constants.Query.redirect)`. otherwise, the `initial` state wouldn't work.

Reply:
These are not assumptions, but actually how the authorization URL works, as it requires authentication and we are working with a web view that is not authenticated automatically. 

Anyway, I added comments explaining the tracking for documentation in db1738aba056907c493d9f86af58e21e10e2f807.